optimum/exporters/onnx/base.py

```diff
@@ -473,13 +474,80 @@ def flatten_output_collection_property(self, name: str, field: Iterable[Any]) ->
         return flattened_output
 
 
+class ConfigBehavior(str, enum.Enum):
+    """
+    Specifies the behavior of the [`~exporters.onnx.base.OnnxSeq2SeqConfigWithPast`]:
+        - MONOLITH: the config can be used to export the whole seq2seq model as a single file.
+        - ENCODER: the config can be used to export the encoder part of the seq2seq model.
+        - DECODER: the config can be used to export the decoder part of the seq2seq model.
+    """
+
+    MONOLITH = "monolith"
+    ENCODER = "encoder"
+    DECODER = "decoder"
+
+
 class OnnxSeq2SeqConfigWithPast(OnnxConfigWithPast):
     """
     Inherits from [`~exporters.onnx.OnnxConfigWithPast`]. A base class to handle the ONNX configuration of encoder-decoder models.
     """
 
     PAD_ATTENTION_MASK_TO_MATCH_TOTAL_SEQUENCE_LENGTH = False
 
+    def __init__(
+        self,
+        config: "PretrainedConfig",
+        task: str = "default",
+        patching_specs: Optional[List[PatchingSpec]] = None,
+        use_past: bool = False,
```

Comment:
this is somewhat nitpick, but I could not find a docstring for what `use_past`, `use_past_in_inputs`, `use_present_in_outputs` mean and do.

We should add one somewhere in my opinion, otherwise (at least I) you have to guess and try to see what it means. Do you think it is useful? It can be left to an other PR.

Reply:
We did not really document those configs in details in terms of their attributes.
I will do that in another PR (I might also try to find some better way than having `use_past` and the other two, which kind of collide).

Reply:
awesome!

Reply:
Created an issue => #595